Mathlib/Dynamics/TopologicalEntropy/CoverEntropy.lean

```diff
@@ -0,0 +1,590 @@
+      simp only [Finset.coe_sort_coe, mem_iUnion, Subtype.exists, exists_prop] at this
+      rcases this with ⟨z, z_s, hz⟩
+      exact ⟨⟨z, z_s⟩, hz⟩
+    choose! t ht using key
+    specialize h_dyncover t
+    simp only [Finset.coe_sort_coe, mem_iUnion, Subtype.exists, toFinset_range,
+      Finset.mem_image, Finset.mem_univ, true_and, exists_prop, exists_exists_eq_and, sn]
+    simp only [mem_preimage, Subtype.forall, Finset.mem_range] at ht
+    use dyncover t
+    simp only [Finset.coe_image, Finset.coe_univ, image_univ, mem_range, exists_apply_eq_apply,
+      true_and]
+    apply h_dyncover
+    simp only [mem_iInter, mem_preimage, Finset.mem_range]
+    exact ht
+  · rw [toFinset_card]
+    apply le_trans (Fintype.card_range_le dyncover)
+    simp only [Fintype.card_fun, Fintype.card_coe, Fintype.card_fin, le_refl]
+
+lemma exists_isDynCoverOf_of_isCompact_uniformContinuous [UniformSpace X] {T : X → X} {F : Set X}
+    (F_comp : IsCompact F) (h : UniformContinuous T) {U : Set (X × X)} (U_uni : U ∈ 𝓤 X) (n : ℕ) :
+    ∃ s : Finset X, IsDynCoverOf T F U n s := by
+  have uni_ite := dynEntourage_mem_uniformity h U_uni n
+  let open_cover := fun x : X ↦ ball x (dynEntourage T U n)
+  have := IsCompact.elim_nhds_subcover F_comp open_cover (fun (x : X) _ ↦ ball_mem_nhds x uni_ite)
+  rcases this with ⟨s, _, s_cover⟩
+  exact ⟨s, s_cover⟩
+
+lemma exists_isDynCoverOf_of_isCompact_invariant [UniformSpace X] {T : X → X} {F : Set X}
+    (F_comp : IsCompact F) (F_inv : MapsTo T F F) {U : Set (X × X)} (U_uni : U ∈ 𝓤 X) (n : ℕ) :
+    ∃ s : Finset X, IsDynCoverOf T F U n s := by
+  rcases comp_symm_mem_uniformity_sets U_uni with ⟨V, V_uni, V_symm, V_U⟩
+  let open_cover := fun x : X ↦ ball x V
+  have := IsCompact.elim_nhds_subcover F_comp open_cover (fun (x : X) _ ↦ ball_mem_nhds x V_uni)
+  rcases this with ⟨s, _, s_cover⟩
+  have : IsDynCoverOf T F V 1 s := by
+    intro x x_F
+    specialize s_cover x_F
+    simp only [mem_iUnion, exists_prop] at s_cover
+    rcases s_cover with ⟨y, y_s, _⟩
+    simp only [dynEntourage, Nat.lt_one_iff, iInter_iInter_eq_left, Function.iterate_zero,
+      Prod.map_id, preimage_id_eq, id_eq, mem_iUnion]
+    use y, y_s
+  rcases this.iterate_le_pow F_inv V_symm n with ⟨t, t_dyncover, t_card⟩
+  rw [one_mul n] at t_dyncover
+  exact ⟨t, t_dyncover.of_entourage_subset V_U⟩
+
+/-! ### Minimal cardinal of dynamical covers -/
+
+/-- The smallest cardinal of a `(U, n)`-dynamical cover of `F`. Takes values in `ℕ∞`, and is
+  infinite if and only if `F` admits no finite dynamical cover.-/
+noncomputable def coverMincard (T : X → X) (F : Set X) (U : Set (X × X)) (n : ℕ) : ℕ∞ :=
+  ⨅ (s : Finset X) (_ : IsDynCoverOf T F U n s), (s.card : ℕ∞)
+
+lemma coverMincard_le_card {T : X → X} {F : Set X} {U : Set (X × X)} {n : ℕ} {s : Finset X}
+    (h : IsDynCoverOf T F U n s) :
+    coverMincard T F U n ≤ s.card := iInf₂_le s h
+
+lemma coverMincard_monotone_time (T : X → X) (F : Set X) (U : Set (X × X)) :
```

Comment:
```suggestion
lemma coverMincard_monotone (T : X → X) (F : Set X) (U : Set (X × X)) :
```

Reply:
Done, thanks.

Reply:
Reverting this one.

There is a statement in another file (but under the same namespace) which goes

`Monotone (fun F : Set X ↦ coverMincard T F U n)`

I know that `time` is not a good descriptor, but I need a way to differentiate these two statements.